### PR TITLE
Fix deprecated ObjectMixin method

### PR DESCRIPTION
**Pull request by janatjak**

### CODE REVIEW

File: src/Kdyby/Doctrine/Connection.php [hunk 1]
```diff
@@ -143,12 +143,7 @@ public function insert($tableExpression, array $data, array $types = [])
 
 
 	/**
-	 * @param string $query
-	 * @param array $params
-	 * @param array $types
-	 * @param \Doctrine\DBAL\Cache\QueryCacheProfile $qcp
-	 * @return \Doctrine\DBAL\Driver\Statement
-	 * @throws DBALException
+     * {@inheritdoc}
```
Comment (enumag) [inline src/Kdyby/Doctrine/Connection.php]: wrong indentation

File: src/Kdyby/Doctrine/Diagnostics/Panel.php [hunk 2]
```diff
@@ -388,8 +388,7 @@ public static function renderException($e, Nette\DI\Container $dic)
 
 			if ($invalidTable = Strings::match($e->getMessage(), '~table \'(.*?)\'~i')) {
 				foreach ($em->getMetadataFactory()->getAllMetadata() as $class) {
-					/** @var Kdyby\Doctrine\Mapping\ClassMetadata $class */
-					if ($class->getTableName() === $invalidTable[1]) {
+					if ($class instanceof Doctrine\ORM\Mapping\ClassMetadata && $class->getTableName() === $invalidTable[1]) {
```
Comment (enumag) [inline src/Kdyby/Doctrine/Diagnostics/Panel.php]: I don't understand why phpstan wants this. `Kdyby\Doctrine\Mapping\ClassMetadata` extends `Doctrine\ORM\Mapping\ClassMetadata` so the method should exist, right? Do you know what's up here?

File: src/Kdyby/Doctrine/Entities/MagicAccessors.php [hunk 3]
```diff
@@ -261,7 +260,7 @@ public function __call($name, $args)
 	 */
 	public static function __callStatic($name, $args)
 	{
-		ObjectMixin::callStatic(get_called_class(), $name, $args);
+		ObjectHelpers::strictStaticCall(get_called_class(), $name);
```
Comment (enumag) [inline src/Kdyby/Doctrine/Entities/MagicAccessors.php]: ObjectHelpers is not available in 2.4. We need to update composer.json.

File: src/Kdyby/Doctrine/Entities/MagicAccessors.php [hunk 4]
```diff
@@ -428,7 +427,10 @@ public function __isset($name)
 	 */
 	public function __unset($name)
 	{
-		ObjectMixin::remove($this, $name);
+        $class = get_class($this);
+        if (!ObjectHelpers::hasProperty($class, $name)) {
+            throw new Nette\MemberAccessException("Cannot unset the property $class::\$$name.");
+        }
```
Comment (enumag) [inline src/Kdyby/Doctrine/Entities/MagicAccessors.php]: Wrong indentation.

File: src/Kdyby/Doctrine/Diagnostics/Panel.php (hunk 2 above)
Comment (janatjak) [reply src/Kdyby/Doctrine/Diagnostics/Panel.php]: phpstan 0.9 does not want to work with doctype $class

**Comment by janatjak** [comment]
fixed

**Comment by enumag** [comment]
Thanks!

**Pull request merged**